src/scheduler/AsyncTask.php

```diff
@@ -170,8 +171,14 @@ public function publishProgress(mixed $progress) : void{
 	 * @internal Only call from AsyncPool.php on the main thread
 	 */
 	public function checkProgressUpdates() : void{
-		while(($progress = $this->progressUpdates->shift()) !== null){
-			$this->onProgressUpdate(igbinary_unserialize($progress));
+		$timing = Timings::getAsyncTasksProgressUpdateTimings($this);
+		$timing->startTiming();
+		try{
+			while(($progress = $this->progressUpdates->shift()) !== null){
+				$this->onProgressUpdate(igbinary_unserialize($progress));
+			}
+		} finally{
+			$timing->stopTiming();
```

Comment:
why put the timings here? seems like they would be more at home in the AsyncPool directly